R/utilities-figures.R

```diff
@@ -223,3 +223,160 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
 
   return(exportConfiguration)
 }
+
+#' Generate plots as defined in excel file `projectConfiguration$plotsFile`
+#'
+#' @param simulatedScenarios A list of simulated scenarios as returned by `runScenarios()`
+#' @param observedData A list of `DataSet` objects
+#' @param projectConfiguration Object of class `ProjectConfiguration`
+#' that contains information about the output paths and the excel file
+#' where plots are defined.
+#' @param stopIfNotFound If TRUE (default), the function stops if any of the
+#' simulated results or observed data are not found. If FALSE a warning is printed.
+#'
+#'
+#' @return A list of `ggplot` objects
+#'
+#' @export
+createPlotsFromExcel <- function(simulatedScenarios, observedData, projectConfiguration, stopIfNotFound = TRUE) {
+  validateIsOfType(observedData, "DataSet", nullAllowed = TRUE)
+  validateIsOfType(projectConfiguration, "ProjectConfiguration")
+  dfDataCombined <- readExcel(projectConfiguration$plotsFile, sheet = "DataCombined")
```

Comment:
It should be `dfDataCombined <- readExcel(file.path(projectConfiguration$paramsFolder, projectConfiguration$plotsFile), sheet = "DataCombined")`